homeassistant/components/huawei_lte/select.py

```diff
@@ -0,0 +1,115 @@
+"""Support for Huawei LTE selects."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from functools import partial
+import logging
+
+from huawei_lte_api.enums.net import LTEBandEnum, NetworkBandEnum
+
+from homeassistant.components.select import (
+    DOMAIN as SELECT_DOMAIN,
+    SelectEntity,
+    SelectEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import HuaweiLteBaseEntityWithDevice
+from .const import DOMAIN, KEY_NET_NET_MODE, NETWORKMODE_TO_STRING
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up from config entry."""
+    router = hass.data[DOMAIN].routers[config_entry.entry_id]
+    selects: list[Entity] = []
+
+    desc = SelectEntityDescription(
+        key="network_mode", name="Preferred mode", entity_category=EntityCategory.CONFIG
+    )
+    selects.append(
+        HuaweiLteBaseSelect(
+            router,
+            entity_description=desc,
+            key=KEY_NET_NET_MODE,
+            item="NetworkMode",
+            choices=NETWORKMODE_TO_STRING,
+            setter_fn=partial(
+                router.client.net.set_net_mode,
+                LTEBandEnum.ALL,
+                NetworkBandEnum.ALL,
+            ),
+        )
+    )
+
+    async_add_entities(selects, True)
+
+
+@dataclass
+class HuaweiLteBaseSelect(HuaweiLteBaseEntityWithDevice, SelectEntity):
+    """Huawei LTE select base class."""
+
+    entity_description: SelectEntityDescription
+    key: str
+    item: str
+    choices: dict[str, str]
+    setter_fn: Callable
```

Comment:
This should probably be moved into a custom entity description to be consistent with other platforms.